helpers/QueryHelper.php

```diff
@@ -471,58 +494,172 @@ public static function table_exists( $table ) {
 	 *
 	 * @return mixed  Based on output param, default OBJECT.
 	 */
-	public static function get_joined_data(string $primary_table, array $joining_tables, array $select_columns, array $where = [], string $order_by = '', $limit = 10, $offset = 0, string $order = 'DESC', string $output = 'OBJECT') {
+	public static function get_joined_data(
+		string $primary_table,
+		array $joining_tables,
+		array $select_columns,
+		array $where = [],
+		array $search = [],
+		string $order_by = '',
+		$limit = 10,
+		$offset = 0,
+		string $order = 'DESC',
+		string $output = 'OBJECT'
+	) {
 		global $wpdb;
 
-		$obj = new self();
-		
 		$select_clause = implode(', ', $select_columns);
-		
+
 		$from_clause = $primary_table;
-		
+
 		$join_clauses = '';
 		foreach ($joining_tables as $relation) {
 			$join_clauses .= " {$relation['type']} JOIN {$relation['table']} ON {$relation['on']}";
 		}
-		
-		$where_clause = !empty($where) ? 'WHERE ' . $obj->build_where_clause($where) : '';
-		
+
+		$where_clause = !empty($where) ? 'WHERE ' . self::build_where_clause($where) : '';
+
+		if (!empty($search)) {
+			$search_clause = self::build_like_clause( $search );
+			// foreach ($search as $column => $value) {
+			// 	$search_clauses[] = $wpdb->prepare("{$column} LIKE %s", '%' . $wpdb->esc_like($value) . '%');
+			// }
```

Comment:
Remove this if this is not required. If you keep this for reference then add a `//@TODO: comment` for that.